File: app/buildOLD/tmp/kapt3/stubs/debug/com/chapelotas/app/data/PreferencesRepositoryImpl.java
package com.chapelotas.app.data;

/**
 * Implementación del repositorio de preferencias usando SharedPreferences
 */
@javax.inject.Singleton()
@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000V\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0002\b\u0003\n\u0002\u0010\u000b\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\b\n\u0002\u0010\u000e\n\u0002\b\u0003\n\u0002\u0010\b\n\u0002\b\u0005\n\u0002\u0010\"\n\u0002\u0010\t\n\u0002\b\t\b\u0007\u0018\u0000 /2\u00020\u0001:\u0001/B\u0011\b\u0007\u0012\b\b\u0001\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u000e\u0010\n\u001a\u00020\u000bH\u0096@\u00a2\u0006\u0002\u0010\fJ\u000e\u0010\r\u001a\u00020\u0007H\u0096@\u00a2\u0006\u0002\u0010\fJ\u000e\u0010\u000e\u001a\u00020\u000fH\u0096@\u00a2\u0006\u0002\u0010\fJ\b\u0010\u0010\u001a\u00020\u0007H\u0002J\u000e\u0010\u0011\u001a\u00020\u000bH\u0096@\u00a2\u0006\u0002\u0010\fJ\u000e\u0010\u0012\u001a\b\u0012\u0004\u0012\u00020\u00070\u0013H\u0016J\u000e\u0010\u0014\u001a\u00020\u000bH\u0096@\u00a2\u0006\u0002\u0010\fJ\u0010\u0010\u0015\u001a\u00020\u000b2\u0006\u0010\u0016\u001a\u00020\u0007H\u0002J\u0016\u0010\u0017\u001a\u00020\u000b2\u0006\u0010\u0018\u001a\u00020\u000fH\u0096@\u00a2\u0006\u0002\u0010\u0019J\u0016\u0010\u001a\u001a\u00020\u000b2\u0006\u0010\u001b\u001a\u00020\u001cH\u0096@\u00a2\u0006\u0002\u0010\u001dJ\u001e\u0010\u001e\u001a\u00020\u000b2\u0006\u0010\u001f\u001a\u00020 2\u0006\u0010!\u001a\u00020 H\u0096@\u00a2\u0006\u0002\u0010\"J\u0016\u0010#\u001a\u00020\u000b2\u0006\u0010\u001b\u001a\u00020\u001cH\u0096@\u00a2\u0006\u0002\u0010\u001dJ\u001c\u0010$\u001a\u00020\u000b2\f\u0010%\u001a\b\u0012\u0004\u0012\u00020\'0&H\u0096@\u00a2\u0006\u0002\u0010(J\u0016\u0010)\u001a\u00020\u000b2\u0006\u0010*\u001a\u00020 H\u0096@\u00a2\u0006\u0002\u0010+J\u001e\u0010,\u001a\u00020\u000b2\u0006\u0010\u001f\u001a\u00020 2\u0006\u0010!\u001a\u00020 H\u0096@\u00a2\u0006\u0002\u0010\"J\u0016\u0010-\u001a\u00020\u000b2\u0006\u0010\u0016\u001a\u00020\u0007H\u0096@\u00a2\u0006\u0002\u0010.R\u0014\u0010\u0005\u001a\b\u0012\u0004\u0012\u00020\u00070\u0006X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\b\u001a\u00020\tX\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u00060"}, d2 = {"Lcom/chapelotas/app/data/PreferencesRepositoryImpl;", "Lcom/chapelotas/app/domain/repositories/PreferencesRepository;", "context", "Landroid/content/Context;", "(Landroid/content/Context;)V", "_preferencesFlow", "Lkotlinx/coroutines/flow/MutableStateFlow;", "Lcom/chapelotas/app/domain/entities/UserPreferences;", "prefs", "Landroid/content/SharedPreferences;", "acceptPrivacyPolicy", "", "(Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "getUserPreferences", "isFirstTimeUser", "", "loadPreferences", "markAsExperiencedUser", "observeUserPreferences", "Lkotlinx/coroutines/flow/Flow;", "resetToDefaults", "savePreferences", "preferences", "setSarcasticMode", "enabled", "(ZLkotlin/coroutines/Continuation;)Ljava/lang/Object;", "updateCriticalAlertSound", "soundUri", "", "(Ljava/lang/String;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "updateDailySummaryTime", "hour", "", "minute", "(IILkotlin/coroutines/Continuation;)Ljava/lang/Object;", "updateNotificationSound", "updatePreferredCalendars", "calendarIds", "", "", "(Ljava/util/Set;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "updateReminderMinutesBefore", "minutes", "(ILkotlin/coroutines/Continuation;)Ljava/lang/Object;", "updateTomorrowSummaryTime", "updateUserPreferences", "(Lcom/chapelotas/app/domain/entities/UserPreferences;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "Companion", "app_debug"})
public final class PreferencesRepositoryImpl implements com.chapelotas.app.domain.repositories.PreferencesRepository {
    @org.jetbrains.annotations.NotNull()
    private final android.content.Context context = null;
    @org.jetbrains.annotations.NotNull()
    private final android.content.SharedPreferences prefs = null;
    @org.jetbrains.annotations.NotNull()
    private final kotlinx.coroutines.flow.MutableStateFlow<com.chapelotas.app.domain.entities.UserPreferences> _preferencesFlow = null;
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String PREFS_NAME = "chapelotas_preferences";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_DAILY_SUMMARY_HOUR = "daily_summary_hour";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_DAILY_SUMMARY_MINUTE = "daily_summary_minute";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_TOMORROW_SUMMARY_HOUR = "tomorrow_summary_hour";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_TOMORROW_SUMMARY_MINUTE = "tomorrow_summary_minute";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_SARCASTIC_MODE = "sarcastic_mode";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_CRITICAL_ALERT_SOUND = "critical_alert_sound";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_NOTIFICATION_SOUND = "notification_sound";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_REMINDER_MINUTES = "reminder_minutes_before";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_PRIVACY_ACCEPTED = "privacy_policy_accepted";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_FIRST_TIME_USER = "is_first_time_user";
    @org.jetbrains.annotations.NotNull()
    private static final java.lang.String KEY_PREFERRED_CALENDARS = "preferred_calendars";
    @org.jetbrains.annotations.NotNull()
    public static final com.chapelotas.app.data.PreferencesRepositoryImpl.Companion Companion = null;
    
    @javax.inject.Inject()
    public PreferencesRepositoryImpl(@dagger.hilt.android.qualifiers.ApplicationContext()
    @org.jetbrains.annotations.NotNull()
    android.content.Context context) {
        super();
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object getUserPreferences(@org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super com.chapelotas.app.domain.entities.UserPreferences> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.NotNull()
    public kotlinx.coroutines.flow.Flow<com.chapelotas.app.domain.entities.UserPreferences> observeUserPreferences() {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updateUserPreferences(@org.jetbrains.annotations.NotNull()
    com.chapelotas.app.domain.entities.UserPreferences preferences, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updateDailySummaryTime(int hour, int minute, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updateTomorrowSummaryTime(int hour, int minute, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object setSarcasticMode(boolean enabled, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updatePreferredCalendars(@org.jetbrains.annotations.NotNull()
    java.util.Set<java.lang.Long> calendarIds, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object acceptPrivacyPolicy(@org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object markAsExperiencedUser(@org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updateCriticalAlertSound(@org.jetbrains.annotations.NotNull()
    java.lang.String soundUri, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updateNotificationSound(@org.jetbrains.annotations.NotNull()
    java.lang.String soundUri, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object updateReminderMinutesBefore(int minutes, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object resetToDefaults(@org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    @java.lang.Override()
    @org.jetbrains.annotations.Nullable()
    public java.lang.Object isFirstTimeUser(@org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super java.lang.Boolean> $completion) {
        return null;
    }
    
    /**
     * Carga las preferencias desde SharedPreferences
     */
    private final com.chapelotas.app.domain.entities.UserPreferences loadPreferences() {
        return null;
    }
    
    /**
     * Guarda las preferencias en SharedPreferences
     */
    private final void savePreferences(com.chapelotas.app.domain.entities.UserPreferences preferences) {
    }
    
    @kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000\u0014\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0010\u000e\n\u0002\b\f\b\u0086\u0003\u0018\u00002\u00020\u0001B\u0007\b\u0002\u00a2\u0006\u0002\u0010\u0002R\u000e\u0010\u0003\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0005\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0006\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0007\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\b\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\t\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\n\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u000b\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\f\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\r\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u000e\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u000f\u001a\u00020\u0004X\u0082T\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0010"}, d2 = {"Lcom/chapelotas/app/data/PreferencesRepositoryImpl$Companion;", "", "()V", "KEY_CRITICAL_ALERT_SOUND", "", "KEY_DAILY_SUMMARY_HOUR", "KEY_DAILY_SUMMARY_MINUTE", "KEY_FIRST_TIME_USER", "KEY_NOTIFICATION_SOUND", "KEY_PREFERRED_CALENDARS", "KEY_PRIVACY_ACCEPTED", "KEY_REMINDER_MINUTES", "KEY_SARCASTIC_MODE", "KEY_TOMORROW_SUMMARY_HOUR", "KEY_TOMORROW_SUMMARY_MINUTE", "PREFS_NAME", "app_debug"})
    public static final class Companion {
        
        private Companion() {
            super();
        }
    }
}

File: app/src/main/java/com/chapelotas/app/data/PreferencesRepositoryImpl.kt
package com.chapelotas.app.data

import android.content.Context
import android.content.SharedPreferences
import com.chapelotas.app.domain.entities.UserPreferences
import com.chapelotas.app.domain.repositories.PreferencesRepository
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import java.time.LocalTime
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Implementación del repositorio de preferencias usando SharedPreferences
 */
@Singleton
class PreferencesRepositoryImpl @Inject constructor(
    @ApplicationContext private val context: Context
) : PreferencesRepository {

    private val prefs: SharedPreferences = context.getSharedPreferences(
        PREFS_NAME,
        Context.MODE_PRIVATE
    )

    private val _preferencesFlow = MutableStateFlow(loadPreferences())

    companion object {
        private const val PREFS_NAME = "chapelotas_preferences"

        // Keys
        private const val KEY_DAILY_SUMMARY_HOUR = "daily_summary_hour"
        private const val KEY_DAILY_SUMMARY_MINUTE = "daily_summary_minute"
        private const val KEY_TOMORROW_SUMMARY_HOUR = "tomorrow_summary_hour"
        private const val KEY_TOMORROW_SUMMARY_MINUTE = "tomorrow_summary_minute"
        private const val KEY_SARCASTIC_MODE = "sarcastic_mode"
        private const val KEY_CRITICAL_ALERT_SOUND = "critical_alert_sound"
        private const val KEY_NOTIFICATION_SOUND = "notification_sound"
        private const val KEY_REMINDER_MINUTES = "reminder_minutes_before"
        private const val KEY_PRIVACY_ACCEPTED = "privacy_policy_accepted"
        private const val KEY_FIRST_TIME_USER = "is_first_time_user"
        private const val KEY_PREFERRED_CALENDARS = "preferred_calendars"
    }

    override suspend fun getUserPreferences(): UserPreferences {
        return loadPreferences()
    }

    override fun observeUserPreferences(): Flow<UserPreferences> {
        return _preferencesFlow.asStateFlow()
    }

    override suspend fun updateUserPreferences(preferences: UserPreferences) {
        savePreferences(preferences)
        _preferencesFlow.value = preferences
    }

    override suspend fun updateDailySummaryTime(hour: Int, minute: Int) {
        val current = loadPreferences()
        val updated = current.copy(
            dailySummaryTime = LocalTime.of(hour, minute)
        )
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun updateTomorrowSummaryTime(hour: Int, minute: Int) {
        val current = loadPreferences()
        val updated = current.copy(
            tomorrowSummaryTime = LocalTime.of(hour, minute)
        )
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun setSarcasticMode(enabled: Boolean) {
        val current = loadPreferences()
        val updated = current.copy(isSarcasticModeEnabled = enabled)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun updatePreferredCalendars(calendarIds: Set<Long>) {
        val current = loadPreferences()
        val updated = current.copy(preferredCalendars = calendarIds)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun acceptPrivacyPolicy() {
        val current = loadPreferences()
        val updated = current.copy(hasAcceptedPrivacyPolicy = true)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun markAsExperiencedUser() {
        val current = loadPreferences()
        val updated = current.copy(isFirstTimeUser = false)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun updateCriticalAlertSound(soundUri: String) {
        val current = loadPreferences()
        val updated = current.copy(criticalAlertSound = soundUri)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun updateNotificationSound(soundUri: String) {
        val current = loadPreferences()
        val updated = current.copy(notificationSound = soundUri)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun updateReminderMinutesBefore(minutes: Int) {
        val current = loadPreferences()
        val updated = current.copy(minutesBeforeEventForReminder = minutes)
        savePreferences(updated)
        _preferencesFlow.value = updated
    }

    override suspend fun resetToDefaults() {
        val defaults = UserPreferences()
        savePreferences(defaults)
        _preferencesFlow.value = defaults
    }

    override suspend fun isFirstTimeUser(): Boolean {
        return prefs.getBoolean(KEY_FIRST_TIME_USER, true)
    }

    /**
     * Carga las preferencias desde SharedPreferences
     */
    private fun loadPreferences(): UserPreferences {
        return UserPreferences(
            dailySummaryTime = LocalTime.of(
                prefs.getInt(KEY_DAILY_SUMMARY_HOUR, 7),
                prefs.getInt(KEY_DAILY_SUMMARY_MINUTE, 0)
            ),
            tomorrowSummaryTime = LocalTime.of(
                prefs.getInt(KEY_TOMORROW_SUMMARY_HOUR, 20),
                prefs.getInt(KEY_TOMORROW_SUMMARY_MINUTE, 0)
            ),
            isSarcasticModeEnabled = prefs.getBoolean(KEY_SARCASTIC_MODE, false),
            criticalAlertSound = prefs.getString(KEY_CRITICAL_ALERT_SOUND, "default_ringtone") ?: "default_ringtone",
            notificationSound = prefs.getString(KEY_NOTIFICATION_SOUND, "default_notification") ?: "default_notification",
            minutesBeforeEventForReminder = prefs.getInt(KEY_REMINDER_MINUTES, 15),
            hasAcceptedPrivacyPolicy = prefs.getBoolean(KEY_PRIVACY_ACCEPTED, false),
            isFirstTimeUser = prefs.getBoolean(KEY_FIRST_TIME_USER, true),
            preferredCalendars = prefs.getStringSet(KEY_PREFERRED_CALENDARS, emptySet())
                ?.mapNotNull { it.toLongOrNull() }
                ?.toSet() ?: emptySet()
        )
    }

    /**
     * Guarda las preferencias en SharedPreferences
     */
    private fun savePreferences(preferences: UserPreferences) {
        prefs.edit().apply {
            putInt(KEY_DAILY_SUMMARY_HOUR, preferences.dailySummaryTime.hour)
            putInt(KEY_DAILY_SUMMARY_MINUTE, preferences.dailySummaryTime.minute)
            putInt(KEY_TOMORROW_SUMMARY_HOUR, preferences.tomorrowSummaryTime.hour)
            putInt(KEY_TOMORROW_SUMMARY_MINUTE, preferences.tomorrowSummaryTime.minute)
            putBoolean(KEY_SARCASTIC_MODE, preferences.isSarcasticModeEnabled)
            putString(KEY_CRITICAL_ALERT_SOUND, preferences.criticalAlertSound)
            putString(KEY_NOTIFICATION_SOUND, preferences.notificationSound)
            putInt(KEY_REMINDER_MINUTES, preferences.minutesBeforeEventForReminder)
            putBoolean(KEY_PRIVACY_ACCEPTED, preferences.hasAcceptedPrivacyPolicy)
            putBoolean(KEY_FIRST_TIME_USER, preferences.isFirstTimeUser)
            putStringSet(KEY_PREFERRED_CALENDARS, preferences.preferredCalendars.map { it.toString() }.toSet())
            apply()
        }
    }
}